hs_tools_resource/static/js/toolresource.js

```diff
@@ -22,4 +22,25 @@ $(document).ready(function() {
     if ($('#resourceSpecificTab #id_url').val() != "") {
         $('#tool-icon-preview').attr('src', $('#resourceSpecificTab #id_url').val());
     }
+
+    /*
+    * The following code is only meant to affect resources landing pages other than
+    * Web Apps themselves, as it formats the "Open with..." dropdown button on the
+    * landing page of resources that have a web app associated with them.
+    */
+    var setDefaultIcon = function($this) {
+        var style = $this.hasClass('user-webapp-icon') ? "" : "dropdown-";
+        $this
+            .attr('src', '/static/img/web-app-default-icon.png')
+            .removeClass(style + 'user-webapp-icon')
+            .addClass(style + 'default-webapp-icon');
+    };
+    $('.user-webapp-icon, .dropdown-user-webapp-icon').each(function() {
+        if ($(this).attr('src') == '') {
+            setDefaultIcon($(this));
+        }
+    }).on('error', function() {
+        setDefaultIcon($(this));
+    });
+    $('h2').attr('style', 'width: calc(100% - 110px);');
```

Comment:
This selector is too broad and can create problems in the future. I suggest to add a class to the h2 title tag and use it in this selector.

The selector would then look like this:
> $('h2.resource-title').attr('style', 'width: calc(100% - 110px);');